core/src/components/button/button.ionic.scss

```diff
@@ -203,283 +201,62 @@
-}
-
-:host(.button-block) .button-native::after {
-  clear: both;
-}
-
-// Full Button
-// --------------------------------------------------
-
-:host(.button-full) {
-  display: block;
-}
-
-:host(.button-full) .button-native {
-  @include globals.margin-horizontal(0);
-
-  width: 100%;
-
-  contain: content;
-}
-
-:host(.button-full:not(.button-round)) .button-native {
-  @include globals.border-radius(0);
-
-  border-right-width: 0;
-  border-left-width: 0;
-}
-
-.button-native {
-  @include globals.border-radius(var(--border-radius));
-  @include globals.font-smoothing();
-  @include globals.margin(0);
-  @include globals.padding(var(--padding-top), var(--padding-end), var(--padding-bottom), var(--padding-start));
-  @include globals.text-inherit();
-
-  display: flex;
-
-  position: relative;
-
-  align-items: center;
-
-  width: 100%;
-  height: 100%;
-
-  min-height: inherit;
-
-  transition: var(--transition);
-
-  border-width: var(--border-width);
-  border-style: var(--border-style);
-  border-color: var(--border-color);
-
-  outline: none;
-
-  background: var(--background);
-
-  line-height: 1;
-
-  box-shadow: var(--box-shadow);
-
-  contain: layout style;
-  cursor: pointer;
-
-  opacity: var(--opacity);
-  overflow: var(--overflow);
-
-  z-index: 0;
-  box-sizing: border-box;
-  appearance: none;
-}
-
-.button-native::-moz-focus-inner {
-  border: 0;
-}
-
-.button-inner {
-  display: flex;
-  position: relative;
-
-  flex-flow: row nowrap;
-  flex-shrink: 0;
-  align-items: center;
-  justify-content: center;
-
-  width: 100%;
-  height: 100%;
-
-  z-index: 1;
-}
-
-// Button Slots
-// --------------------------------------------------
-
-::slotted([slot="start"]),
-::slotted([slot="end"]) {
-  flex-shrink: 0;
+  opacity: 0.6;
 }
 
 // Button Icons
 // --------------------------------------------------
 
 ::slotted(ion-icon) {
   font-size: globals.$ionic-font-size-500;
-  pointer-events: none;
-}
-
-// Button: States
-// --------------------------------------------------
-
-.button-native::after {
-  @include globals.button-state();
 }
 
-// Button Focused
-:host(.ion-focused) {
-  color: var(--color-focused);
+::slotted(ion-icon[slot="start"]) {
+  @include globals.margin-horizontal(null, globals.$ionic-space-200);
 }
 
-:host(.ion-focused) .button-native::after {
-  background: var(--background-focused);
-
-  opacity: var(--background-focused-opacity);
-}
-
-// Button Hover
-@media (any-hover: hover) {
-  :host(:hover) {
-    color: var(--color-hover);
-  }
-
-  :host(:hover) .button-native::after {
-    background: var(--background-hover);
-
-    opacity: var(--background-hover-opacity);
-  }
-}
-
-// Button Activated
-:host(.ion-activated) {
-  color: var(--color-activated);
-}
-
-:host(.ion-activated) .button-native::after {
-  background: var(--background-activated);
-
-  opacity: var(--background-activated-opacity);
+::slotted(ion-icon[slot="end"]) {
+  @include globals.margin-horizontal(globals.$ionic-space-200, null);
```

Comment:
Same as my other comment about space and button size

Reply:
https://github.com/ionic-team/ionic-framework/pull/29754/commits/53ce98234b1b216c702ec8f69f4fdd393c5a2edc